horace_core/sqw/PixelData/@PixelData/PixelData.m

```diff
@@ -238,6 +242,7 @@
         %               in-memory data. (Optional)
         %
         obj.object_id_ = polyval(randi([0, 9], 1, 5), 10);
+        obj.page_memory_size_ = get(hor_config, 'pixel_page_size');
```

Comment:
this should be combined with the existing page-size code block which should be moved here from the 'file-backed' section of the constructor below:
```
if exist('mem_alloc', 'var')
    obj.page_memory_size_ = mem_alloc;
else
    obj.page_memory_size_ = get(hor_config, 'pixel_page_size');
end
```

Reply:
There is no validation on the page size when passed as an argument. This should have the same validation you perform on the config option 'memalloc >= onerow, memalloc <= maxArraySize`